portfolio/src/main/webapp/index.js

```diff
@@ -0,0 +1,92 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const FETCH_COMMENTS_URL = '/data';
+const COMMENT_CONTAINER = 'comments-container';
+const COMMENT_CLASS_NAME = 'comment';
+const NAME_CLASS_NAME = 'name-output';
+const MESSAGE_CLASS_NAME = 'message-output';
+const CLASS_TAG = 'class';
+const LIST_CHILD_TAG = 'li';
+const DIV_TAG = 'div';
+const P_TAG = 'p';
+const SMALL_TAG = 'small';
```

Comment:
I wonder if some of these constants are a bit of overkill.
Do you foresee wanting to use  a different HTML tag for paragraphs or divs?
Any thoughts?

Sometimes having literals in code is fine, if they are common literals that you don't anticipate needing to change. The benefit of having defined constants is if they could reasonably be changed, or if you worry about misspelling the contents. The central constant makes changes propagate to all uses and ensure that everything will be spelled the same in each reference (via type-checking of the constant name).

Reply:
That makes sense! I mostly just added them as constants because I assumed it was a good coding practice, but it definitely makes more sense to just keep them as literals.